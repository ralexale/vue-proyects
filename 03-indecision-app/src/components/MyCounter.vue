<template>
  <section class="mt-4 mb-10">
    <h3 class="text-2xl">Counter: {{ counter }}</h3>
    <h3 data-testid="square-label" class="text-2xl">Square: {{ squareCounter }}</h3>
    <!-- <span>{{ text }}</span> -->

    <div class="buttons">
      <button @click="counter++" class="text-white rounded-lg bg-green-500 px-4 text-xl">+1</button>
      <button @click="counter--" class="text-white rounded-lg bg-red-500 px-4 text-xl">-1</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { useCounter } from '@/composables/userCounter';

interface Props {
  value: number;
  // text?: string;
}

const props = defineProps<Props>();

const { counter, squareCounter } = useCounter(props.value);
</script>
