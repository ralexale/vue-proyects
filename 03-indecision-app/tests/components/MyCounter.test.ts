import { mount } from '@vue/test-utils';

import MyCounter from '@/components/MyCounter.vue';

describe('MyCounter', () => {
  test('should compare the counter value with the square value from the snapshot', () => {
    // preparaci贸n
    const wrapper = mount(MyCounter, {
      props: {
        value: 10,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders the counter value and square value properly', () => {
    // preparaci贸n
    const value = 5;
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const [h3Counter, h3Square] = wrapper.findAll('h3');

    // cuando usamos find el primero que encuentre es el que toma
    expect(wrapper.find('h3').text()).toContain(`Counter: ${value}`);
    expect(wrapper.find('[data-testid="square-label"]').text()).toContain(
      `Square: ${value * value}`,
    );

    expect(h3Counter.text()).toContain(`Counter: ${value}`);
    expect(h3Square.text()).toContain(`Square: ${value * value}`);
  });

  test('increments the counter +1 button is clicked', async () => {
    // preparaci贸n
    const value = 7;
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const [h3Counter, h3Square] = wrapper.findAll('h3');

    // estimulo
    const btnIncrement = wrapper.find('button');
    await btnIncrement.trigger('click');

    // comportamiento esperado
    expect(h3Counter.text()).toContain(`Counter: ${value + 1}`);
    expect(h3Square.text()).toContain(`Square: ${(value + 1) * (value + 1)}`);
  });

  test('decrements the counter -2 button is clicked twice', async () => {
    // preparaci贸n
    const value = 7;
    const wrapper = mount(MyCounter, {
      props: {
        value: value,
      },
    });
    const [h3Counter, h3Square] = wrapper.findAll('h3');

    // estimulo
    const [, btnDecrement] = wrapper.findAll('button');
    await btnDecrement.trigger('click');
    await btnDecrement.trigger('click');

    // comportamiento esperado
    expect(h3Counter.text()).toContain(`Counter: ${value - 2}`);
    expect(h3Square.text()).toContain(`Square: ${(value - 2) * (value - 2)}`);
  });
});
