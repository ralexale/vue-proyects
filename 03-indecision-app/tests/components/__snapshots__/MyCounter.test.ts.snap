// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyCounter > should compare the counter value with the square value from the snapshot 1`] = `
"<section class="mt-4 mb-10">
  <h3 class="text-2xl">Counter: 10</h3>
  <h3 data-testid="square-label" class="text-2xl">Square: 100</h3><!-- <span>{{ text }}</span> -->
  <div class="buttons"><button class="text-white rounded-lg bg-green-500 px-4 text-xl">+1</button><button class="text-white rounded-lg bg-red-500 px-4 text-xl">-1</button></div>
</section>"
`;
